### PR TITLE
Adds to docs for styling standards based on using Sass and CSS modules

### PR DESCRIPTION
**Pull request by kimallen**
## Description

Adds documentation for standardizing styling using Sass and CSS Modules, along with some other styling standards.

## Reviewer Notes

Anything else to add?  Is the wording clear?
I'm thinking about switching all `data-cy` references to `data-test` and making that the test attribute.  What do you think?

## Code Review Verification Steps

* [ ] Request review from a member of a different team.

## References

[ADR](https://github.com/transcom/mymove/blob/master/docs/adr/0031-css-tooling.md)
[Improving CSS: Sass and CSS Modules (and more!) within MilMove](https://docs.google.com/document/d/1TZyMg9oNV88fa078dztL1CWqqgXQFcFHMcB226LlN-8/edit)

### CODE REVIEW

File: docs/frontend.md [hunk 1]
```diff
@@ -113,16 +116,46 @@ Adhere to Airbnb's [JavaScript Style Guide](https://github.com/airbnb/javascript
...
+  * [Beginners Guide to Sass](https://coolestguidesontheplanet.com/guide-beginning-sass-css/)
+  * CSS Modules with React: [The Complete Guide](https://blog.yipl.com.np/css-modules-with-react-the-complete-guide-a98737f79c7c)
+
+#### Classnames
+
+* Use [`classnames`](https://github.com/JedWatson/classnames) package for assigning classes based on boolean values
+
+#### rem vs. em
+
+Understand the [difference between rem and em](https://zellwk.com/blog/rem-vs-em/). Which one you use can impact styling elsewhere on the webpage.
```
Comment (jim) [inline docs/frontend.md]: Is there a guideline here as to which to use by default?

File: docs/frontend.md [hunk 2]
```diff
@@ -113,16 +116,46 @@ Adhere to Airbnb's [JavaScript Style Guide](https://github.com/airbnb/javascript
...
+
+#### Using Sass and CSS Modules
+
+* All new components should utilize [Sass](https://sass-lang.com/documentation/file.SASS_REFERENCE.html) and [CSS modules](https://github.com/css-modules/css-modules) (See [ADR](https://github.com/transcom/mymove/blob/master/docs/adr/0031-css-tooling.md)).  To apply Sass and CSS Modules, name files  with this syntax: `<component>.module.scss`
+* Use CSS Modules' [`composes`](https://bambielli.com/til/2017-08-11-css-modules-composes/#) to build a new class out of other pre-defined classes
+* Where to put styles?
+  * For global styles, such as colors and themes, utilize global variables in files such as `colors.scss`
+  * Styles shared across the app should go into `src/shared/styles`.  Most common styles can fit into `common.module.scss`, though additional files may be appropriate.
```
Comment (jim) [inline docs/frontend.md]: A few of these lines uses `module` and some use `modules` in filenames. It's not totally clear which to use when to me.

Comment (kimallen) [reply docs/frontend.md]: oops, it should be `module`.  `modules` doesn't work (I already had to debug that).  Good catch.

File: docs/frontend.md (hunk 1 above)
Comment (kimallen) [reply docs/frontend.md]: From the article, it seems quite complex, and totally depends on the usage.  Not sure which by default would yield the fewest unintended consequences.

Comment (kimallen) [reply docs/frontend.md]: Hopefully as we learn more we can add more specificity about what to use in the docs, but seemed like the article could it explain much better than I could and I didn't see a succinct way to summarize it.

**Comment by kimallen** [comment]
I'll merge this.  Then later when I change `data-cy` to `data-test`, I'll change the documentation and let everyone know.

**Pull request merged**